ouroboros-consensus/src/Ouroboros/Consensus/NodeNetwork.hs

```diff
@@ -259,32 +266,35 @@ protocolCodecsId :: Monad m
                       (AnyMessage (ChainSync blk (Tip blk)))
                       (AnyMessage (LocalTxSubmission (GenTx blk) (ApplyTxErr blk)))
 protocolCodecsId = ProtocolCodecs {
-      pcChainSyncCodec         = codecChainSyncId
-    , pcBlockFetchCodec        = codecBlockFetchId
-    , pcTxSubmissionCodec      = codecTxSubmissionId
-    , pcLocalChainSyncCodec    = codecChainSyncId
-    , pcLocalTxSubmissionCodec = codecLocalTxSubmissionId
+      pcChainSyncCodec            = codecChainSyncId
+    , pcBlockFetchCodec           = codecBlockFetchId
+    , pcBlockFetchCodecSerialised = undefined -- TODO
```

Comment:
TODO

Reply:
Don't we have this now? We can fill this in. Is it not being used anywhere yet?

Reply:
The problem is that the `bytes` (what is sent over the wire) of `pcBlockFetchCodec` and `pcBlockFetchCodecSerialised` must match. This is fine for the real codec (`ByteString`s), but for the `CodecId`, it doesn't work, as we have `AnyMessage (BlockFetch (Serialised blk))` for the server and `AnyMessage (BlockFetch blk)` for the client. In the tests we then create a channel to send such `AnyMessage (BlockFetch blk)` with on one side the server and on the other the client, but the types don't match.

In the tests, I could use `ByteString` as the `bytes` for the BlockFetch protocol, but `protocolCodecsId` still won't type-check unless I add an extra `bytesSBF` type parameter for `pcBlockFetchCodecSerialised` to `ProtocolCodecs`.

Reply:
I have a solution for this in my last revision. It fixes all outstanding TODOs.

The solution works but is probably not the most elegant one.